File: Java/Palindrome.java
import java.util.*;

class Palindrome {
   static boolean isPalindrome(String s) {
      int left = 0;
      int right = s.length() - 1;

      while (left < right) {
         if (s.charAt(left) != s.charAt(right)) {
            return false;
         }
         left++;
         right--;
      }
      return true;
   }

   public static void main(String args[]) {
      System.out.println("Eshaan Vaswani C33 181 \n");

      Scanner s = new Scanner(System.in);

      System.out.println("Enter a string: ");
      String str = s.nextLine();

      if (isPalindrome(str)) {
         System.out.println("The string is a palindrome.");
      } else {
         System.out.println("The string is not a palindrome.");
      }

      s.close();
   }
}

File: CP/palindrome.c
#include<stdio.h>

int main(){
    printf("ESHAAN VASWANI D-34 \n\n");

    int i,j,k=0,c=0;
    char a[100];

    printf("Enter the string : ");
    gets(a);

    for(i=0;a[i]!='\0';i++){
        k++;
    }
    printf("The size of the string is %d \n",k);

    for(j=0;j<=k/2;j++){
        if(a[j]==a[k-1-j])
            c=1;
        else{
            c=0;
            break;
        }
    }

if(c==0)
    printf("It is not a palindrome");
else if(c==1)
    printf("It is a palindrome");

return 0;
}
